addons/modules/functions/fnc_moduleUnpackStaticWeapon.sqf

```diff
@@ -0,0 +1,120 @@
+#include "script_component.hpp"
+/*
+ * Author: Ampersand
+ * Zeus module function to unpack and assemble a static weapon from supported backpacks.
+ *
+ * Arguments:
+ * 0: Logic <OBJECT>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [LOGIC] call zen_modules_fnc_moduleUnpackStaticWeapon
+ *
+ * Public: No
+ */
+
+params ["_logic"];
+
+private _gunner = attachedTo _logic;
+deleteVehicle _logic;
+
+if (isNull _gunner) exitWith {
+    [LSTRING(NoUnitSelected)] call EFUNC(common,showMessage);
+};
+
+if !(_gunner isKindOf "CAManBase") exitWith {
+    [LSTRING(OnlyInfantry)] call EFUNC(common,showMessage);
+};
+
+if !(alive _gunner) exitWith {
+    [LSTRING(OnlyAlive)] call EFUNC(common,showMessage);
+};
+
+if (isPlayer _gunner) exitWith {
+    ["str_a3_cfgvehicles_moduleremotecontrol_f_errorPlayer"] call EFUNC(common,showMessage);
+};
+
+if (units _gunner findIf {isPlayer _x} != -1) exitWith {
+    [LSTRING(ModuleUnpackStaticWeapon_Player)] call EFUNC(common,showMessage);
+};
+
+private _backpack = backpack _gunner;
+if (_backpack isEqualTo "") exitWith {
+    [LSTRING(ModuleUnpackStaticWeapon_Unit)] call EFUNC(common,showMessage);
+};
+
+private _assembleInfo = ["assembleTo", "base", "displayName", "dissasembleTo", "primary"] apply {(configOf backpackContainer _gunner >> "assembleInfo" >> _x) call BIS_fnc_getCfgData};
+_assembleInfo params ["_assembleTo", "_compatibleBases", "_displayName", "_dissasembleTo", "_primary"];
+if (_compatibleBases isEqualType "") then {_compatibleBases = [];};
+
+if (_primary == 1 && {_compatibleBases isEqualTo [] || {_compatibleBases isEqualTo "" || {_compatibleBases isEqualTo [""]}}}) then {
```

Comment:
`_compatibleBases` cannot be a string at this point. Line 50 changes it to an array.